### PR TITLE
Add purge pile

### PR DESCRIPTION
**Pull request by granttitus**

### CODE REVIEW

File: src/game/Actions/Actions.tsx [hunk 1]
```diff
@@ -83,6 +83,23 @@ export const exec = (action: any, state: any) => {
...
+            const owner: Player = getCardOwner(action.cardID, state)
+            const card = removeCardByID(owner, action.cardID)
+            if (card.cardsUnderneath) {
+                card.cardsUnderneath.forEach((cardUnderneath: Creature | Artifact | CardInHand) => {
+                    owner.discardPile.push(cardUnderneath)
+                })
+                card.cardsUnderneath = []
+            }
+            if (card.upgrades) {
+                card.upgrades.forEach((upgrade: Creature | Artifact | CardInHand) => {
```
Comment (Dunkoro) [inline src/game/Actions/Actions.tsx]: I don't understand this line. Does it say that the upgrade that's attached to a creature can itself be a creature, an artifact or a card in hand?

File: src/game/Actions/Actions.tsx [hunk 2]
```diff
@@ -83,6 +83,23 @@ export const exec = (action: any, state: any) => {
             }
             owner.drawPile.unshift(card)
         },
+        [Event.PurgeCard]: () => {
+            const owner: Player = getCardOwner(action.cardID, state)
+            const card = removeCardByID(owner, action.cardID)
+            if (card.cardsUnderneath) {
+                card.cardsUnderneath.forEach((cardUnderneath: Creature | Artifact | CardInHand) => {
```
Comment (Dunkoro) [inline src/game/Actions/Actions.tsx]: I don't understand this line. Does it say that the upgrade that's attached to a creature can itself be a creature, an artifact or a card in hand?

File: src/game/Actions/Actions.tsx (hunk 1 above)
Comment (granttitus) [reply src/game/Actions/Actions.tsx]: Fixed in an upcoming commit

File: src/game/Actions/Actions.tsx (hunk 2 above)
Comment (granttitus) [reply src/game/Actions/Actions.tsx]: Fixed in an upcoming commit

**Pull request merged**